### PR TITLE
Data factory Diagnostic settings policy now goes into compliance

### PR DESCRIPTION
**Pull request by ericvruder**
## Overview/Summary

Added two new log sources in azure data factory diagnostics settings to deploy diagnostic settings policy for azure data factory, SandboxPipelineRuns and SandboxActivityRuns

## This PR fixes/adds/changes/removes

Data factory Diagnostic settings policy was never going into compliance since it was checking for all logs being enabled, but the policy itself was missing two new ones: SandboxPipelineRuns and SandboxActivityRuns. Those have been added to the policy and will now be compliant after a remediation task

### Breaking Changes
There are no breaking changes

## Testing Evidence

image
image

## As part of this Pull Request I have

- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [ ] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [X] Performed testing and provided evidence.
- [X] Updated relevant and associated documentation.
- [X] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

File: src/resources/Microsoft.Authorization/policyDefinitions/Deploy-Diagnostics-DataFactory.json [hunk 1]
```diff
@@ -185,6 +185,14 @@
                         {
                           "category": "SSISIntegrationRuntimeLogs",
                           "enabled": "[[parameters('logsEnabled')]"
+                        },
+                        {
+                          "category": "SandboxPipelineRuns",
+                          "enabled": "[parameters('logsEnabled')]"
```
Comment (jtracey93) [inline src/resources/Microsoft.Authorization/policyDefinitions/Deploy-Diagnostics-DataFactory.json:191]:
```suggestion
                          "enabled": "[[parameters('logsEnabled')]"
```

File: src/resources/Microsoft.Authorization/policyDefinitions/Deploy-Diagnostics-DataFactory.json [hunk 2]
```diff
@@ -185,6 +185,14 @@
...
+                        },
+                        {
+                          "category": "SandboxActivityRuns",
+                          "enabled": "[parameters('logsEnabled')]"
```
Comment (jtracey93) [inline src/resources/Microsoft.Authorization/policyDefinitions/Deploy-Diagnostics-DataFactory.json:195]:
```suggestion
                          "enabled": "[[parameters('logsEnabled')]"
```

File: docs/wiki/Whats-new.md [hunk 3]
```diff
@@ -56,6 +56,7 @@ Here's what's changed in Enterprise Scale/Azure Landing Zones:
 - Updated "**Deploy Diagnostic Settings to Azure Services**" initiative replacing deprecated policy for diagnostic settings on Storage Account
 - Removed all exclusions (parameters) from the Microsoft Cloud Security Benchmark (currently Azure Security Benchmark) initiative assignment to standardize across reference architectures and align with best practice.
 Impacted assignment: Deploy-ASC-Monitoring
+- Updated "**Deploy Diagnostic Settings for Data Factory to Log Analytics workspace" to now go into compliance after a remediation task
```
Comment (jtracey93) [inline docs/wiki/Whats-new.md:59]:
```suggestion
- Updated "**Deploy Diagnostic Settings for Data Factory to Log Analytics workspace" to include new categories of: `SandboxPipelineRuns` & `SandboxActivityRuns`
```

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Thanks @ericvruder for the PR. I have made some suggestions and also we need to increment the policy version to `1.2.0` from `1.1.0` (i will do this in your branch shortly :))

**Review by jtracey93 — APPROVED** [review]
LGTM

**Comment by jtracey93** [comment]
> @ericvruder please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> 
> Contributor License Agreement

@ericvruder can you complete this please as we cannot merge your code until you agree

**Comment by ericvruder** [comment]
@microsoft-github-policy-service agree

**Comment by ericvruder** [comment]
What is the extra square bracket for? Kinda surprised by that. Doesn't look like correct syntax? 😅

**Comment by jtracey93** [comment]
> What is the extra square bracket for? Kinda surprised by that. Doesn't look like correct syntax? 😅

@ericvruder this is due to the way the ALZ portal experience is built, we just need to add some extra ARM escaping to ensure when deployed the actual parameter value is used on policy assignment instead of just a literal string of `[parameters('abc')]`

**Pull request merged**